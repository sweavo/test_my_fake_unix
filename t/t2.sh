#!/bin/bash
## In MSys bash, if you delete a file that another process is holding, windows 
## will like to MSys that the file is deleted, but then you cannot create a new
## file with the same name.

function exit_handler 
{
    echo exit_status $?
    echo done
}
trap exit_handler EXIT

set -e # exit on error

TMPFILE=$(mktemp _run_t2XXXXX)

echo holding
( echo one; sleep 10; echo two ) > $TMPFILE &

sleep 1
echo removing
rm $TMPFILE 
echo "rm returned $?"
sleep 1
echo rewriting
echo hi > $TMPFILE
echo "echo with redirect returned $?"

