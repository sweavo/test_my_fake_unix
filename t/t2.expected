holding
removing
rm returned 0
rewriting
echo with redirect returned 0
exit_status 0
done
